core/modules/rproc/InfileMerger.cc

```diff
@@ -284,18 +263,23 @@ bool InfileMerger::merge(std::shared_ptr<proto::WorkerResponse> response) {
 
 
     auto start = std::chrono::system_clock::now();
-    // &&& ret = _applyMysql(infileStatement);
-    ret = _applyMysqlNew(infileStatement, infileMgr, virtFile);
+    if (_dbEngine == MYISAM) {
+        ret = _applyMysqlMyIsam(infileStatement);
+    } else { // Used for both INNODB and MEMORY
+        ret = _applyMysqlInnoDb(infileStatement);
+    }
```

Comment:
I think using `case` here would be future-proof, in case if more engine options would be used. For others throw an exception:
```
switch (_dbEngine) {
    case MYISAM:
        ret = _applyMysqlMyIsam(infileStatement);
        break;
    case INNODB:
    case MEMORY:
        ret = _applyMysqlInnoDb(infileStatement);
        break;
    default:
        throw ...
}```